### PR TITLE
Reduced usage of Internal code

### PR DESCRIPTION
**Pull request by pranaysashank**

### CODE REVIEW

File: examples/Examples.hs [hunk 1]
```diff
@@ -19,18 +19,22 @@ import Streamly.Internal.Data.Unfold (Unfold)
...
+import qualified Streamly.FileSystem.Handle as FH
+import qualified Streamly.Unicode.Stream as U
 import qualified Streamly.Network.Socket as SK
 import qualified Streamly.Network.Inet.TCP as TCP
 
 import qualified Streamly.Internal.Data.Stream.IsStream as S
-import qualified Streamly.Internal.Unicode.Stream as U
-import qualified Streamly.Internal.Data.Fold as FL
+       (intercalateSuffix)
+import qualified Streamly.Internal.Unicode.Stream as U (words, unwords)
```
Comment (harendra-kumar) [inline examples/Examples.hs:29]: We can replace these with their definitions and plan to expose `Stream.interpose` in 0.8.0.

File: examples/Examples.hs [hunk 2]
```diff
@@ -19,18 +19,22 @@ import Streamly.Internal.Data.Unfold (Unfold)
...
+import qualified Streamly.Internal.Data.Fold as FL (mkPure, classify)
 import qualified Streamly.Internal.Data.Unfold as UF
-import qualified Streamly.Internal.FileSystem.Dir as Dir
+       (fromList, fold, enumerateFromToIntegral, identity, outerProduct, map)
```
Comment (harendra-kumar) [inline examples/Examples.hs:32]: We can expose these in 0.8.0.

File: examples/Examples.hs [hunk 3]
```diff
@@ -19,18 +19,22 @@ import Streamly.Internal.Data.Unfold (Unfold)
...
+import qualified Streamly.Internal.FileSystem.Dir as Dir (toFiles)
 import qualified Streamly.Internal.FileSystem.File as File
-import qualified Streamly.Internal.FileSystem.Handle as FH
-import qualified Streamly.Internal.Network.Socket as SK
-import qualified Streamly.Internal.Data.Stream.Parallel as Par
+       (toChunks, fromChunks, toBytes, fromBytes, write, read)
+import qualified Streamly.Internal.Network.Socket as SK (writeStrings)
```
Comment (harendra-kumar) [inline examples/Examples.hs:36]: We can expand this to its definition and expose everything required for that definition.

File: examples/Examples.hs [hunk 4]
```diff
@@ -19,18 +19,22 @@ import Streamly.Internal.Data.Unfold (Unfold)
...
 import qualified Streamly.Data.Array.Storable.Foreign as A
+import qualified Streamly.FileSystem.Handle as FH
+import qualified Streamly.Unicode.Stream as U
 import qualified Streamly.Network.Socket as SK
 import qualified Streamly.Network.Inet.TCP as TCP
 
 import qualified Streamly.Internal.Data.Stream.IsStream as S
-import qualified Streamly.Internal.Unicode.Stream as U
-import qualified Streamly.Internal.Data.Fold as FL
+       (intercalateSuffix)
```
Comment (harendra-kumar) [inline examples/Examples.hs:28]: We can expose this in 0.8.0.

File: examples/FromFileClient.hs [hunk 5]
```diff
@@ -6,8 +6,8 @@
 import System.Environment (getArgs)
 
 import qualified Streamly.Prelude as S
-import qualified Streamly.Internal.FileSystem.Handle as IFH
-import qualified Streamly.Internal.Network.Inet.TCP as TCP
+import qualified Streamly.FileSystem.Handle as FH
+import qualified Streamly.Internal.Network.Inet.TCP as TCP (writeChunks)
```
Comment (harendra-kumar) [inline examples/FromFileClient.hs:10]: We can expose this in 0.8.0.

File: examples/HandleIO.hs [hunk 6]
```diff
@@ -6,13 +6,14 @@ import qualified Streamly.Data.Fold as FL
...
+import qualified Streamly.Unicode.Stream as US
 import qualified Streamly.Prelude as S
 -- import qualified Streamly.FileSystem.FD as FH
 
-import qualified Streamly.Internal.Data.Fold as FL
-import qualified Streamly.Internal.Unicode.Stream as US
-import qualified Streamly.Internal.Memory.ArrayStream as AS
-import qualified Streamly.Internal.Data.Stream.IsStream as S
+import qualified Streamly.Internal.Data.Fold as FL (classify)
+import qualified Streamly.Internal.Unicode.Stream as US (lines)
```
Comment (harendra-kumar) [inline examples/HandleIO.hs:14]: `lines` can be expanded to its definition.

File: examples/Rate.hs [hunk 7]
```diff
@@ -1,10 +1,10 @@
 import qualified Streamly.Prelude as S
-import qualified Streamly.Internal.Data.Stream.IsStream as Internal
+import qualified Streamly.Internal.Data.Stream.IsStream as S (timestamped)
```
Comment (harendra-kumar) [inline examples/Rate.hs:2]: We can expose this in 0.8.0.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]
Can we use full names of the modules in imports e.g. Stream instead of S, Fold instead of FL etc.

File: examples/WordClassifier.hs [hunk 8]
```diff
@@ -16,19 +16,20 @@ import qualified Data.List as List
...
 import qualified Streamly.Unicode.Stream as S
-import qualified Streamly.Internal.Unicode.Stream as S
+import qualified Streamly.Internal.Unicode.Stream as S (words)
 import qualified Streamly.Data.Fold as FL
-import qualified Streamly.Internal.Data.Fold as IFL
-import qualified Streamly.Internal.Data.Unfold as IUF
-import qualified Streamly.Internal.FileSystem.File as File
+import qualified Streamly.Internal.Data.Fold as FL
+       (rollingHash, rollingHashWithSalt)
+import qualified Streamly.Internal.Data.Unfold as UF (fold)
```
[the remark this answers is not in the document]
Comment (pranaysashank) [reply examples/WordClassifier.hs:23]: We can't use Stream.fold without using more internal functions

File: examples/WordClassifier.hs [hunk 9]
```diff
@@ -16,19 +16,20 @@ import qualified Data.List as List
 import qualified Data.Ord as Ord
 import           Foreign.Storable (Storable(..))
 import qualified Streamly.Unicode.Stream as S
-import qualified Streamly.Internal.Unicode.Stream as S
+import qualified Streamly.Internal.Unicode.Stream as S (words)
```
[the remark this answers is not in the document]
Comment (pranaysashank) [reply examples/WordClassifier.hs]: We need isSpace to inline for good performance, perhaps we can copy its definition in the example files.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]
We can move `isAlpha`, and `toLower` to `Streamly.Internal.Unicode.Char` module. Also, `isSpace`.

We can keep the usage of `words` if `isSpace` does not inline. Perf needs to remain good.

Raise an issue for proposed streamly changes.

**Pull request merged**